### PR TITLE
[Stacked] Add dialogue verifier

### PR DESCRIPTION
**Pull request by ferozco**
## Before this PR
We did not test the generated client against the spec verifier

## After this PR
We do

### CODE REVIEW

File: conjure-java-client-verifier/src/test/java/com/palantir/conjure/java/compliance/AutoDeserializeTest.java [hunk 1]
```diff
@@ -41,7 +42,9 @@
     public static final VerificationServerRule server = new VerificationServerRule();
 
     private static final Logger log = LoggerFactory.getLogger(AutoDeserializeTest.class);
-    private static final AutoDeserializeService testService = VerificationClients.autoDeserializeService(server);
+    private static final AutoDeserializeService jerseyTestService = VerificationClients.autoDeserializeService(server);
+    private static final BlockingAutoDeserializeService dialogueTestService =
```
Comment (iamdanfox) [inline conjure-java-client-verifier/src/test/java/com/palantir/conjure/java/compliance/AutoDeserializeTest.java:46]: comment to justify that this probably gives us good enough coverage of the async one too?

File: conjure-java-client-verifier/src/test/java/com/palantir/conjure/java/compliance/SingleParamServicesTest.java [hunk 2]
```diff
@@ -44,42 +44,61 @@
...
+                    VerificationClients.dialogueSinglePathParamService(server))
+            .putAll(
+                    "singleHeaderService",
+                    VerificationClients.singleHeaderService(server),
+                    VerificationClients.dialogueSingleHeaderService(server))
+            .putAll(
+                    "singleQueryParamService",
+                    VerificationClients.singleQueryParamService(server),
+                    VerificationClients.dialogueSingleQueryParamService(server))
+            .build();
```
Comment (iamdanfox) [inline conjure-java-client-verifier/src/test/java/com/palantir/conjure/java/compliance/SingleParamServicesTest.java:60]: could we move this var into the `data()` func so it's clear we're only indexing into the multimap (which is a little gross) to build up the testcases?

File: conjure-java-client-verifier/src/test/java/com/palantir/conjure/java/compliance/SingleParamServicesTest.java [hunk 3]
```diff
@@ -91,7 +110,7 @@ public void runTestCase() throws Exception {
 
         System.out.println(String.format("Invoking %s %s(%s)", serviceName, endpointName, jsonString));
 
-        Object service = servicesMaps.get(serviceName);
+        Object service = servicesMaps.get(serviceName).asList().get(serviceIndex);
```
Comment (iamdanfox) [inline conjure-java-client-verifier/src/test/java/com/palantir/conjure/java/compliance/SingleParamServicesTest.java:113]: the .get() would change to just accessing one of the fields provided as a param

File: conjure-java-client-verifier/src/test/java/com/palantir/conjure/java/compliance/VerificationClients.java [hunk 4]
```diff
@@ -18,16 +18,30 @@
...
 import com.palantir.conjure.verification.server.SinglePathParamService;
 import com.palantir.conjure.verification.server.SingleQueryParamService;
+import com.palantir.dialogue.JavaChannels;
+import com.palantir.tritium.metrics.registry.DefaultTaggedMetricRegistry;
 
 public final class VerificationClients {
     private VerificationClients() {}
 
+    private static final DefaultConjureRuntime DEFAULT_CONJURE_RUNTIME =
+            DefaultConjureRuntime.builder().build();
```
Comment (iamdanfox) [inline conjure-java-client-verifier/src/test/java/com/palantir/conjure/java/compliance/VerificationClients.java:43]: I wonder if at some point we could dedupe the undertow & dialogue serializer registries?

File: gradle/verifier.gradle [hunk 5]
```diff
@@ -85,6 +89,16 @@ task conjureJavaJersey(type: JavaExec) {
...
 
+task conjureJavaDialogue(type: JavaExec) {
+    main = "com.palantir.conjure.java.cli.ConjureJavaCli"
+    classpath = project(':conjure-java').sourceSets.main.runtimeClasspath
+    args 'generate',  "${-> configurations.verificationApi.singleFile}", 'src/generatedDialogue/java', '--dialogue'
+
+    inputs.file "${-> configurations.verificationApi.singleFile}"
+    outputs.dir "src/generatedDialogue/java"
+    doFirst { delete "src/generatedDialogue/java/*" }
+}
```
Comment (iamdanfox) [inline gradle/verifier.gradle]: can we hook this up to the first-class local codegen now? configuration containing the tgz file with a nice `builtBy` should work here?

**Review by iamdanfox — APPROVED** [review]

File: conjure-java-client-verifier/src/test/java/com/palantir/conjure/java/compliance/SingleParamServicesTest.java (hunk 2 above)
Comment (ferozco) [reply conjure-java-client-verifier/src/test/java/com/palantir/conjure/java/compliance/SingleParamServicesTest.java:60]: looks like we actually rely on having the full name: https://github.com/palantir/conjure-java/pull/758/files#diff-99ee67231f23433ff3738d89c722d01dR125 🤦‍♂

File: conjure-java-client-verifier/src/test/java/com/palantir/conjure/java/compliance/VerificationClients.java (hunk 4 above)
Comment (carterkozak) [reply conjure-java-client-verifier/src/test/java/com/palantir/conjure/java/compliance/VerificationClients.java:43]: I don't think we should, they cover similar concepts for separate uses so we would lose flexibility. I don't think they're large or complex enough for that to be worthwhile.

**Pull request merged**